cleaner/podman.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2022.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleaner
+
+import (
+	"context"
+	"os"
+
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/sirupsen/logrus"
+)
+
+// A valid URI connection should be scheme://
+// For example tcp://localhost:<port>
+// or unix:///run/podman/podman.sock
+// or ssh://<user>@<host>[:port]/run/podman/podman.sock?secure=True
+func (p Podman) GetRemoteConnection(uri string, identity string) (context.Context, error) {
+	conn, err := bindings.NewConnectionWithIdentity(context.Background(), uri, identity)
+	if err != nil {
+		logrus.Errorf("%s \n", err)
+		return nil, err
+	}
+	return conn, err
+}
+
+func (p Podman) GetConnection() (context.Context, error) {
+	// ROOTLESS access
+	sockDir := os.Getenv("XDG_RUNTIME_DIR")
+	socket := "unix:" + sockDir + "/podman/podman.sock"
+	conn, err := bindings.NewConnection(context.Background(), socket)
+	if err != nil {
+		logrus.Errorf("%s \n", err)
+		return nil, err
+	}
+	return conn, err
+}
+
+func (p Podman) GetImages(connection context.Context) []*entities.ImageSummary {
+	images, err := images.List(connection, nil)
+	if err != nil {
+		logrus.Errorf("%s \n", err)
+		return nil
+	}
+	return images
+}
+
+func (p Podman) PruneImages(connection context.Context) int {
+	options := new(images.PruneOptions).WithAll(true)
+	results, err := images.Prune(connection, options)
+	if err != nil {
+		logrus.Errorf("%s \n", err)
+		return 0
+	}
+	return len(results)
+}
+
+func (p Podman) RemoveImage(name string, force bool, connection context.Context) (*entities.ImageRemoveReport, []error) {
+	return images.Remove(
+		connection,
+		[]string{name},
+		new(images.RemoveOptions).WithForce(force))
+}
+
+func (p Podman) RemoveImages(names []string, connection context.Context) (*entities.ImageRemoveReport, []error) {
```

Comment:
tyo avoid code duplication we can have just one remove method with this signature:

```
func (p Podman) RemoveImages(names []string, force bool, connection context.Context) (*entities.ImageRemoveReport, []error) {
```

`force` should always defaults to `false`.

Reply:
ok